submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmitWebHandler.kt

```diff
@@ -2,122 +2,116 @@ package ac.uk.ebi.biostd.submission.web.handlers
     }
 
-    private fun buildRequest(request: FileSubmitWebRequest): SubmitRequest {
-        val (_, submitter, attrs) = request.submissionConfig
-        val (files, preferredSources) = request.filesConfig
-        val sub = serializationService.deserializeSubmission(request.submission)
-        val extSub = extSubmissionService.findExtendedSubmission(sub.accNo)
-        requireNotProcessing(sub.accNo)
-
-        val source = fileSourcesService.submissionSources(
-            FileSourcesRequest(
-                submitter = submitter,
-                files = files,
-                onBehalfUser = request.onBehalfRequest?.let { onBehalfUtils.getOnBehalfUser(it) },
-                rootPath = sub.rootPath,
-                submission = extSub,
-                preferredSources = preferredSources
-            )
-        )
-        val submission = withAttributes(submission(request.submission, source), attrs)
-
-        userFilesService.uploadFile(submitter, DIRECT_UPLOAD_PATH, request.submission)
-
-        return SubmitRequest(
-            submission = submission,
-            submitter = submitter,
-            onBehalfUser = request.onBehalfRequest?.let { onBehalfUtils.getOnBehalfUser(it) },
-            sources = source,
-            method = FILE,
-        )
+    fun submit(request: FileSubmitWebRequest): Submission {
+        val rqt = buildRequest(request)
+        userFilesService.uploadFile(request.config.submitter, DIRECT_UPLOAD_PATH, request.submission)
+        val extSubmission = subService.submit(rqt)
+        return toSubmissionMapper.toSimpleSubmission(extSubmission)
     }
 
-    private fun withAttributes(submission: Submission, attrs: List<ExtAttributeDetail>): Submission {
-        attrs.forEach { submission[it.name] = it.value }
-        return submission
+    fun submitAsync(request: ContentSubmitWebRequest) {
+        val rqt = buildRequest(request)
+        subService.submitAsync(rqt)
     }
 
-    private fun submission(
-        content: String,
-        format: SubFormat,
-        source: FileSourcesList
-    ) = serializationService.deserializeSubmission(content, format, source)
+    fun submitAsync(request: FileSubmitWebRequest) {
+        val rqt = buildRequest(request)
+        userFilesService.uploadFile(request.config.submitter, DIRECT_UPLOAD_PATH, request.submission)
+        subService.submitAsync(rqt)
+    }
 
-    private fun submission(
-        subFile: File,
-        source: FileSourcesList
-    ) = serializationService.deserializeSubmission(subFile, source)
+    private fun buildRequest(rqt: SubmitWebRequest): SubmitRequest {
+        val (submitter, attrs) = rqt.config
+        val (files, preferredSources) = rqt.filesConfig
+        val onBehalfUser = rqt.onBehalfRequest?.let { onBehalfUtils.getOnBehalfUser(it) }
+
+        /**
+         * Deserialize the submission without considering files.
+         */
+        fun submissionAttributes(): Pair<String, String?> {
```

Comment:
I think the "without considering files" part of the doc for this function is a bit confusing I would suggest: "Deserialize the submission to get the accNo and rootPath attributes. This deserialization doesn't consider the files since they're not needed yet", something like that

I also think, after reviewing this piece of code, that maybe we should improve the names in the serialization service instead of overloading the methods which would make them clearer.